Automattic/wp-calypso
# Site Settings: Fix message  "%siteDomain has been deleted" in notices.

This PR makes the notices be shown when the SITE_DELETE_RECEIVE action is dispatched, instead of doing it on SITE_DELETE_SUCCESS dispatch so the notices middleware can have access to its domain for showing it in the notice before it gets removed from the state.

Before this, we were showing the notice on SITE_DELETE_SUCCESS and the domain was not there in the state.

#### Testing instructions

1. Create a new site.
1. Delete it, and expect the notices "sitedomain is being deleted" and "sitedomain has been deleted" to both show the old site's domain.
